src/core/server/metrics/metrics_service.ts

```diff
@@ -83,5 +92,33 @@ export class MetricsService
       clearInterval(this.collectInterval);
     }
     this.metrics$.complete();
+    this.subscriptions.forEach((subscription) => {
+      subscription.unsubscribe();
+    });
+  }
+
+  private convertToMetricEvent(metrics: OpsMetrics): OpsMetricsEvent {
```

Comment:
@pgayvallet extracting how we report the ops metrics into a function was a great idea! It also gives us some flexibility to change the data, as @afharo wanted.

Reply:
Looks good! I think I would extract that to a dedicated function/file for testability though

Reply:
Looks good! I think I would extract that to a dedicated function/file for testability though

Reply:
Looks good! I think I would extract that to a dedicated function/file for testability though